transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -222,6 +222,38 @@ def _read_ff_dump_resp(self, length):
         flash_dump = self._read_memory(buffer_addr, length)
         return flash_dump
 
+    def _read_mapped_memory(self, device_info, region, read_start_addr, read_length):
+        if read_length is None:
+            if region == 'ram':
+                memory = self._read_memory(device_info.ram_start, device_info.ram_size)
+            elif region == 'flash':
+                memory = self._read_memory(device_info.flash_start, device_info.flash_size)
+            elif region == 'mapped':
+                raise ArgumentError("Must specify a data length to read from mapped memory.")
+        else:
+            if region == 'ram':
+                if read_length > device_info.ram_size:
+                    raise ArgumentError("Invalid data length to read from RAM.")
```

Comment:
I would recommend including the ram size in this string (and the other errors you're raising in this method) so that the user learns why you are rejecting their request.  You can just attach an extra keyword arg like `ram_size=device_info.ram_size`.

Reply:
Fixed